common/views/components/ModalMoreFilters/ModalMoreFilters.tsx

```diff
@@ -0,0 +1,359 @@
+import {
+  CatalogueAggregations,
+  CatalogueAggregationBucket,
+} from '@weco/common/model/catalogue';
+import Space from '../styled/Space';
+import CheckboxRadio from '@weco/common/views/components/CheckboxRadio/CheckboxRadio';
+import { searchFilterCheckBox } from '../../../text/arial-labels';
+import {
+  getAggregationFilterByName,
+  getAggregationRadioGroup,
+} from '@weco/common/utils/filters';
+import RadioGroup from '@weco/common/views/components/RadioGroup/RadioGroup';
+import NextLink from 'next/link';
+import { worksLink } from '../../../services/catalogue/routes';
+import ButtonSolid, {
+  ButtonTypes,
+} from '@weco/common/views/components/ButtonSolid/ButtonSolid';
+import { WorksRouteProps } from '@weco/common/services/catalogue/ts_routes';
+
+type BaseModalProps = {
+  width?: string | null;
+};
+
+type sharedFiltersProps = {
+  changeHandler: () => void;
+  languagesInUrl: string[];
+  genresInUrl: string;
+  subjectsInUrl: string;
+  filtersToShow: string[];
+};
+type MoreFiltersProps = {
+  id: string;
+  showMoreFiltersModal: boolean;
+  setMoreFiltersModal: (arg: boolean) => void;
+  openMoreFiltersButtonRef: RefObject<HTMLInputElement>;
+  aggregations: CatalogueAggregations | undefined;
+  filtersToShow: string[];
+  worksRouteProps: WorksRouteProps;
+  isEnhanced: boolean;
+} & sharedFiltersProps;
+
+type RenderMoreFiltersProps = {
+  genresFilter: CatalogueAggregationBucket[];
+  languagesFilter: CatalogueAggregationBucket[];
+  subjectsFilter: CatalogueAggregationBucket[];
+} & sharedFiltersProps;
+
+const ModalInner = styled.div`
+  display: flex;
+  flex-direction: column;
+  min-width: 320px;
+  max-width: 600px;
+  ${props => props.theme.media.large`
+    width: 600px;
+    top: 10px;
+  `}
+  position: relative;
+  top: 15px;
+  overflow-y: auto;
+  max-height: 80vh;
+`;
+
+// shared styles
+const FilterSection = styled(Space).attrs({
+  h: { size: 'l', properties: ['padding-left', 'padding-right'] },
+  v: { size: 'l', properties: ['padding-top', 'padding-bottom'] },
+  className: classNames({
+    'border-bottom-width-1 border-color-pumice': true,
+  }),
+})``;
+
+const FiltersFooter = styled(Space).attrs({
+  h: { size: 'l', properties: ['padding-left', 'padding-right'] },
+  v: { size: 'l', properties: ['padding-top', 'padding-bottom'] },
+  className: classNames({
+    'bg-white border-color-pumice border-top-width-1 flex flex--v-center flex--h-space-between': true,
+  }),
+})`
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  height: 60px;
+`;
+
+const FiltersHeader = styled(Space).attrs({
+  h: { size: 'm', properties: ['padding-left', 'padding-right'] },
+  v: { size: 'm', properties: ['padding-top', 'padding-bottom'] },
+  className: classNames({
+    'border-color-pumice border-bottom-width-1 absolute': true,
+  }),
+})`
+  text-align: center;
+  top: 0px;
+  left: 0px;
+  width: 100%;
+`;
+
+const OverrideModalWindow = styled(Space).attrs<BaseModalProps>({
```

Comment:
Why do we need this? It scares me as then whenever we change the Modal, we need to make sure that we test the override too.

If we _really_ need this, could it be part of the modal component?

Reply:
we use this to override the overflow so we can have the overflow inside and allow the header and bottom footer fixed. 
I will put this into the modal component as you mentioned and pass props to request the type.